assets/src/higher-order-components/local-storage/use-local-storage.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * External imports
+ */
+import { useState, useEffect } from '@wordpress/element';
+
+const useLocalStorage = ( key ) => {
```

Comment:
Rather than rolling our own implementation, I was thinking we could use something like `https://github.com/WordPress/gutenberg/tree/master/packages/data/src/plugins/persistence` to simply persist the filter-state store between sessions.